cli_tools/common/utils/logging/service/logger.go

```diff
@@ -381,6 +386,34 @@ func (l *Logger) constructLogRequest(logExtension *ComputeImageToolsLogExtension
 	return reqStr, err
 }
 
+// safeRunner supports running a function that may panic. If there's a panic, execution is
+// recovered and the panic's details are captured.
+type safeRunner struct {
```

Comment:
panicSafeRunner? this would make it clear what the runner is making the inner safe from.

Reply:
sgtm, done